addons/hr_recruitment_survey/wizard/recruitment_survey_invite.py

```diff
@@ -0,0 +1,312 @@
+                    invite.subject = "Interview Invitation"
+
+    def _send_mail(self, answer):
+        mail = super()._send_mail(answer)
+        if answer.applicant_id:
+            safe_body = Markup("<div>%s</div>") % (mail.body_html or "")
+            answer.applicant_id.message_post(body=safe_body)
+            mail.send()
+        return mail
+
+    # Security
+    @api.depends('survey_ids.restrict_user_ids')
+    def _compute_accessible_by_current_user(self):
+        current_user = self.env.user
+        for record in self:
+            record.accessible_by_current_user = all(
+                not survey.restrict_user_ids or current_user in survey.restrict_user_ids
+                for survey in record.survey_ids
+            )
+
+    @api.depends('survey_ids.survey_type')
+    def _compute_all_surveys_are_recruitment(self):
+        for record in self:
+            record.all_surveys_are_recruitment = all(
+                survey.survey_type == 'recruitment'
+                for survey in record.survey_ids
+            )
+
+    @api.depends('survey_ids.hr_job_ids.interviewer_ids', 'survey_ids.hr_job_ids.application_ids.interviewer_ids')
+    def _compute_accessible_by_interviewer_user(self):
+        current_user = self.env.user
+        for record in self:
+            record.accessible_by_interviewer_user = all(
+                any(
+                    current_user in (job.interviewer_ids | job.application_ids.interviewer_ids)
+                    for job in survey.hr_job_ids
+                )
+                for survey in record.survey_ids
+            )
+
+    def _search_accessible_by_current_user(self, operator, value):
+        if operator != 'in':
+            return NotImplemented
+
+        current_user = self.env.user
+
+        accessible_survey_ids = self.env['survey.survey']._search([
+            '|',
+            ('restrict_user_ids', '=', False),
+            ('restrict_user_ids', 'any', [('id', '=', current_user.id)])
+        ])
+
+        inaccessible_self_records = self.env['hr.recruitment.survey.invite']._search([
+            ('survey_ids', '!=', False),
+            ('survey_ids', 'any', [('id', 'not in', accessible_survey_ids)])  # True if one ore more survey(s) are NOT accessible
+        ])
+
+        domain = [('id', 'not in', inaccessible_self_records)] if value else [('id', 'in', inaccessible_self_records)]
+        return domain
+
+    def _search_all_surveys_are_recruitment(self, operator, value):
+        if operator != 'in':
+            return NotImplemented
+
+        recruitment_survey_ids = self.env['survey.survey']._search([
+            ('survey_type', '=', 'recruitment'),
+        ])
+
+        inaccessible_self_records = self.env['hr.recruitment.survey.invite']._search([
+            ('survey_ids', '!=', False),
+            ('survey_ids', 'any', [('id', 'not in', recruitment_survey_ids)])  # returns True if one or more survey(s) are NOT of type recruitment
+        ])
```

Comment:
Similar here, I'm 90% sure you can just write this as one search.
```suggestion
        inaccessible_self_records = self.env['hr.recruitment.survey.invite']._search([ 
        "&", 
        "!", ("survey_ids", "any", [
               "&", 
                  ("survey_type", "not in", ["recruitment"]), 
                  ("survey_type", "!=", False)
               ]), 
              ("survey_ids", "!=", False)])
```